hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -222,9 +211,29 @@ public boolean dropTable(TableIdentifier identifier, boolean purge) {
 
   @Override
   public void renameTable(TableIdentifier from, TableIdentifier originalTo) {
-    if (!isValidIdentifier(from)) {
-      throw new NoSuchTableException("Invalid identifier: %s", from);
-    }
+    renameContent(from, originalTo, HiveOperationsBase.ContentType.TABLE);
+  }
+
+  private List<TableIdentifier> listIcebergTables(
+      List<String> tableNames, Namespace namespace, String tableTypeProp)
+      throws TException, InterruptedException {
+    List<Table> tableObjects =
+        clients.run(client -> client.getTableObjectsByName(namespace.level(0), tableNames));
+    return tableObjects.stream()
+        .filter(
+            table ->
+                table.getParameters() != null
+                    && tableTypeProp.equalsIgnoreCase(
+                        table.getParameters().get(BaseMetastoreTableOperations.TABLE_TYPE_PROP)))
+        .map(table -> TableIdentifier.of(namespace, table.getTableName()))
+        .collect(Collectors.toList());
+  }
+
+  private void renameContent(
+      TableIdentifier from,
+      TableIdentifier originalTo,
+      HiveOperationsBase.ContentType contentType) {
+    Preconditions.checkArgument(isValidIdentifier(from), "Invalid identifier: %s", from);
```

Comment:
this is changing existing behavior. Previously it would throw `NoSuchTableException`

Reply:
Intent here was to have common exception for `Table/View`. But I think with `View` impl we will have to revisit this.